docs/tutorial/index.md

```diff
@@ -1,609 +1,553 @@
 
-### Perform a custom action
-
-In Juju, performing most major operations looks the same for every charm. However, charmers sometimes also define additional operations specific to a given charm. These operations are called actions and often have to do with accessing an application deployed by a charm, creating a backup, etc. Below, use the `postgresql-k8s` charm's `set-password` action to generate a password for the default, `operator` username, then use the username and password to access the PostgreSQL application.
+# Output should show microk8s.
+# If it doesn't: Exit the VM (`exit`), re-enter it (`multipass shell my-juju-vm`), then try again.
 
+```
 
+Ensure also that the client has access to Charmhub by launching a random query, e.g., "ingress", and asking for more information about one of the results it shows:
 
-First, get:
+```{terminal}
+:user: ubuntu
+:host: my-juju-vm
+:input: juju find ingress
 
-- the host IP address of the PostgreSQL unit: retrieve it from `juju status` or `juju show-unit` (in the sample outputs above, `10.1.32.152`);
+# Output should show all the charms or charm bundles related to this query that are available on Charmhub.
+# For best results always double-check Charmhub.
 
--  a PostgreSQL username and password: we can use the internal, default user called `operator` and set a password for it using the `set-password` action. Sample session:
+:input: juju info traefik-k8s
 
-```
-juju run postgresql-k8s/leader set-password username=operator password=mysecretpass
+# Output should show name, publisher, quick description, integration endpoints, etc.
+# For the full features and more info see directly the charm's page on Charmhub.
 ```
 
+Our `juju` client is ready! Sneak a peek at what it can do (`juju help commands` -- you can pipe a query to zoom in on feature: `juju help commands | grep application`). Check status (`juju status` and `juju debug-log` -- not much to see there yet). Keep going...
 
-Now, use this information to access the PostgreSQL application:
-
-First, ssh into the PostgreSQL unit (= Kubernetes container). Sample session:
+```{tip}
 
-```text
-ubuntu@my-juju-vm:~$ juju ssh --container postgresql postgresql-k8s/leader bash
-root@postgresql-k8s-0:/#
-```
+Split your terminal window into 3 (or open 3 terminal windows). In all, access your Multipass VM shell (`multipass shell my-juju-vm`) and then:
 
-Verify that `psql` is already installed. Sample session:
+**Shell 1:** Keep using it as you've already been doing so far, namely to type the commands in this tutorial.
 
-```text
-root@postgresql-k8s-0:/# psql --version
-psql (PostgreSQL) 14.10 (Ubuntu 14.10-0ubuntu0.22.04.1)
-```
+**Shell 2:** Run `juju status --relations --color --watch 1s` to watch your deployment status evolve. Things are all right if your `App Status` and your `Unit - Workload` reach `active` and your `Unit - Agent` reaches `idle`. To exit and return to the terminal prompt, press {kbd}`mod` + {kbd}`C` (e.g., {kbd}`Ctrl`+{kbd}`C`).
```

Comment:
```suggestion
**Shell 2:** Run `watch --color juju status --color` to watch your deployment status evolve. Things are all right if your `App Status` and your `Unit - Workload` reach `active` and your `Unit - Agent` reaches `idle`. To exit and return to the terminal prompt, press {kbd}`mod` + {kbd}`C` (e.g., {kbd}`Ctrl`+{kbd}`C`).
```

This is very much 3.x ism, we dropped the `--watch` flag in 4.0. So with one change, you could make it both 3.x and 4.x.

`watch --color juju status --color `

Also, it might be worth raising a bug against juju about the fact that we need the second `--color`. We should be able to detect that it can use color.